service/src/test/java/bio/terra/cbas/controllers/util/TestRunSetHelper.java

```diff
@@ -0,0 +1,373 @@
+                    400, "ApiException thrown for testing purposes.")));
+
+    when(runDao.getRuns(new RunDao.RunsFilters(runSetId, List.of(QUEUED))))
+        .thenReturn(List.of(run1, run2));
+
+    mockRunSetHelper.triggerWorkflowSubmission(
+        runSetRequest, runSet, recordIdToRunIdMapping, mockToken, mockWorkflowUrl);
+
+    // verify that both Runs were set to Error state with correct error message
+    verify(runDao)
+        .updateRunStatusWithError(
+            eq(runId1),
+            eq(CbasRunStatus.SYSTEM_ERROR),
+            any(),
+            eq(
+                "Error while fetching WDS Records for Record ID(s): {MY_RECORD_ID_2=ApiException thrown for testing purposes.}"));
+    verify(runDao)
+        .updateRunStatusWithError(
+            eq(runId2),
+            eq(CbasRunStatus.SYSTEM_ERROR),
+            any(),
+            eq(
+                "Error while fetching WDS Records for Record ID(s): {MY_RECORD_ID_2=ApiException thrown for testing purposes.}"));
+
+    // verify that RunSet was set to Error state
+    verify(runSetDao)
+        .updateStateAndRunSetDetails(eq(runSetId), eq(CbasRunSetStatus.ERROR), eq(2), eq(2), any());
+  }
+
+  @Test
+  void inputCoercionErrorDuringSubmission() throws Exception {
+    final int recordAttributeValueInt1 = 100;
+    final int recordAttributeValueInt2 = 200;
+    RecordAttributes incorrectRecordAttributes1 = new RecordAttributes();
+    incorrectRecordAttributes1.put(recordAttribute1, recordAttributeValueInt1);
+    RecordAttributes incorrectRecordAttributes2 = new RecordAttributes();
+    incorrectRecordAttributes2.put(recordAttribute1, recordAttributeValueInt2);
+
+    // Set up WDS API responses
+    when(wdsService.getRecord(eq(recordType), eq(recordId1), any()))
+        .thenReturn(
+            new RecordResponse()
+                .type(recordType)
+                .id(recordId1)
+                .attributes(incorrectRecordAttributes1));
+    when(wdsService.getRecord(eq(recordType), eq(recordId2), any()))
+        .thenReturn(
+            new RecordResponse()
+                .type(recordType)
+                .id(recordId2)
+                .attributes(incorrectRecordAttributes2));
+
+    when(cbasApiConfiguration.getMaxWorkflowsInBatch()).thenReturn(10);
+    when(uuidSource.generateUUID()).thenReturn(UUID.randomUUID()).thenReturn(UUID.randomUUID());
+
+    mockRunSetHelper.triggerWorkflowSubmission(
+        runSetRequest, runSet, recordIdToRunIdMapping, mockToken, mockWorkflowUrl);
+
+    // verify Runs were set to Error state
+    verify(runDao)
+        .updateRunStatusWithError(
+            eq(runId1),
+            eq(CbasRunStatus.SYSTEM_ERROR),
+            any(),
+            eq(
+                "Input generation failed for record MY_RECORD_ID_1. Coercion error: Coercion from Integer to String failed for parameter myworkflow.mycall.inputname2. Coercion not supported between these types."));
```

Comment:
Is this text specific to the `RunSetHelper`? Maybe we could set up one of the mock function calls to throw a random error and make sure the trigger function sends it to the DB properly? 

Though, I do know this class needs a refactor to break the "trigger" into many sub-functions, is this specific request too much to unpick right now?

Reply:
No this error is not specific to `RunSetsHelper`. The test case here is testing that correct DAO methods are called when there is an error during inputs generation while submitting workflows.